L2/hardfork_devnet_test.md

```diff
@@ -0,0 +1,95 @@
+
+## 1. Launch the devnet with L2Blob and SGT disabled
+Deploy the OP Stack according to the instructions provided [here](https://github.com/QuarkChain/pm/blob/main/L2/opup_devnet_test.md).
+
+### Notes:
+  - When editing `intent.toml`, ensure you remove the following fields to disable the respective features during initial deployment.
+    - `soulGasTokenTimeOffset` 
+    - `l2GenesisBlobTimeOffset`
+### Validation after deployment:
+  - **SGT Validation**: Executing "Spend SGT without native gas token" should result in:
+    ```bash
+    Error: Failed to estimate gas: server returned an error response: error code -32000: gas required exceeds allowance (0)
+    ```
+  - **L2Blob Validation**: Transactions involving L2Blob should yield:
+    ```bash
+    Error: server returned an error response: error code -32000: transaction type not supported
+    ```
+## 2. Updating the Superchain Registry
+Perform the following steps to update the superchain registry:
+  - Clone and checkout the devnet branch
+    ```bash
+    git clone https://github.com/QuarkChain/superchain-registry.git
+    cd superchain-registry
+    git checkout devnet
+    ```
+  - Update configuration files:
+      - Modify `./chainList.toml` and `./chainList.json` based on your deployment specifics.
+      - Edit `./superchain/config/sepolia/qkc.toml`
+          - Set `l2_blob_time` and `soul_gas_token_time` accordingly
+          - Update values in `genesis.l1`, `genesis.l2` and `genesis.system_config`
+          - Ensure all relevant roles and addresses are correctly set.
+  - Generate and compress genesis file:
+    - Download genesis.json into your working directory and rename it:
+      ```bash
+      mv genesis.json qkc.json
+      ```
+    - Install compression tool if necessary:
+      ```bash
+      apt install zstd
+      ```
+    - Compress genesis file using the provided dictionary:
+      ```bash
+      zstd -D ./superchain/extra/dictionary qkc.json
+      ```    
+    - Move compressed file to correct location:
+      ```bash
+      mv qkc.json.zst ./superchain/extra/genesis/sepolia/
+      ```        
+  - Commit and push changes:
+    ```bash
+    git add .
+    git commit -m "Update genesis and config for L2Blob and SGT enabling"
+    git push origin devnet    
+    ```  
+## 3. Generating the Latest `superchain-configs.zip`
+We will generate the zip file on the deployment machine directly.
+  - Modify `op-geth` source:
+    - Navigate to the `op-geth` folder.
+    - Update repository URL in `sync-superchain.sh`
+      ```bash
+      - https://github.com/ethereum-optimism/superchain-registry.git
+      + https://github.com/QuarkChain/superchain-registry.git      
+      ```
+    - Update commit hash in `superchain-registry-commit.txt` to match the latest commit on the devnet branch.
+  - Run `sync-superchain.sh` to generate the new `superchain-configs.zip`
+  - Rebuild binaries: 
+    - For op-geth: 
+      ```bash
+      make geth
+      ```
+    - For op-node: 
```

Comment:
May need to specify the location of superchain-configs.zip

Reply:
fix in https://github.com/QuarkChain/pm/pull/64/commits/260de3e5b0a36afbd9021d9da581caaef15ec755